src/types/__test__/1/collection.test.ts

```diff
@@ -785,9 +828,73 @@ describe('update pinned', () => {
   })
 })
 
+describe('get collection by article id', () => {
+  let server: Awaited<ReturnType<typeof testClient>>
+  let collectionId: string
+
+  beforeAll(async () => {
+    server = await testClient({ isAuth: true, connections })
+    const { data } = await server.executeOperation({
+      query: PUT_COLLECTION,
+      variables: { input: { title: 'test' } },
+    })
+    collectionId = data?.putCollection?.id
+    await server.executeOperation({
+      query: ADD_COLLECTIONS_ARTICLES,
+      variables: {
+        input: {
+          collections: [collectionId],
+          articles: [articleGlobalId1, articleGlobalId4],
+        },
+      },
+    })
+  })
+  test('get collection by article id', async () => {
+    const { data } = await server.executeOperation({
+      query: GET_COLLECTION_BY_ARTICLES,
+      variables: {
+        input1: { id: collectionId },
+        input2: { articleId: articleGlobalId1 },
+      },
+    })
+    expect(data?.node?.id).toBe(collectionId)
+    expect(data?.node?.articles?.edges?.[0]?.node?.id).toBe(articleGlobalId4)
+    expect(data?.node?.articles?.edges?.[1]?.node?.id).toBe(articleGlobalId1)
+  })
+  test('get collection by article id with invalid article id', async () => {
+    const { data } = await server.executeOperation({
+      query: GET_COLLECTION_BY_ARTICLES,
+      variables: {
+        input1: { id: collectionId },
+        input2: { articleId: 'invalid' },
+      },
+    })
+    expect(data?.node).toBeNull()
```

Comment:
to match the error we can use the returned `errors`, e.g. https://github.com/thematters/matters-server/blob/develop/src/types/__test__/1/collection.test.ts#L221